Libraries/libbom/Sources/bom.c

```diff
@@ -248,14 +248,19 @@ bom_index_reserve(struct bom_context *context, size_t count)
     struct bom_header *header = (struct bom_header *)context->memory.data;
 
     /* Insert space for extra indexes at the end of the currently allocated space. */
-    uint32_t index_point = ntohl(header->index_offset) + ntohl(header->index_length);
-    size_t new_index_length = ntohl(header->index_length) + sizeof(struct bom_index) * count;
+    size_t old_index_length = ntohl(header->index_length);
+    uint32_t index_point = ntohl(header->index_offset) + old_index_length;
     ptrdiff_t index_delta = sizeof(struct bom_index) * count;
+    size_t new_index_length = old_index_length + index_delta;
```

Comment:
just caching values to temp vars?
I don't think there is any logic changes here, right?

Reply:
No logic change.  I just changed the variable names so that they were more clear to me while working on this (also, I needed access to different variables for the memset() below).